api/src/test/java/bio/terra/tanagra/app/EntitiesFilterApiControllerTest.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.tanagra.app;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.tanagra.app.controller.EntitiesFiltersApiController;
+import bio.terra.tanagra.generated.model.ApiAttributeValue;
+import bio.terra.tanagra.generated.model.ApiAttributeVariable;
+import bio.terra.tanagra.generated.model.ApiBinaryFilter;
+import bio.terra.tanagra.generated.model.ApiBinaryFilterOperator;
+import bio.terra.tanagra.generated.model.ApiEntityFilter;
+import bio.terra.tanagra.generated.model.ApiFilter;
+import bio.terra.tanagra.generated.model.ApiSqlQuery;
+import bio.terra.tanagra.service.underlay.NauticalUnderlayUtils;
+import bio.terra.tanagra.testing.BaseSpringUnitTest;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("nautical")
+public class EntitiesFilterApiControllerTest extends BaseSpringUnitTest {
+  @Autowired private EntitiesFiltersApiController apiController;
+
+  @Test
+  void generateSqlQuery() {
+    ApiEntityFilter apiEntityFilter =
+        new ApiEntityFilter()
+            .entityVariable("s")
+            .filter(
+                new ApiFilter()
+                    .binaryFilter(
+                        new ApiBinaryFilter()
+                            .attributeVariable(
+                                new ApiAttributeVariable().variable("s").name("rating"))
+                            .operator(ApiBinaryFilterOperator.EQUALS)
+                            .attributeValue(new ApiAttributeValue().int64Val(42L))));
+
+    ResponseEntity<ApiSqlQuery> response =
+        apiController.generateSqlQuery(
+            NauticalUnderlayUtils.NAUTICAL_UNDERLAY_NAME, "sailors", apiEntityFilter);
+    assertEquals(HttpStatus.OK, response.getStatusCode());
+    assertEquals(
+        "SELECT s.s_id FROM my-project-id.nautical.sailors AS s WHERE s.rating = 42",
```

Comment:
Unrelated to this PR: shouldn't `my-project-id.nautical.sailors` be backticked for BigQuery?

Reply:
I don't think it's necessary for this example. e.g. I haven't had any issues with `fc-aou-cdr-synth-test.test_R2019q4r3.cb_criteria` without backticks. https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#identifiers calls out that `-` is fine without backticks.
Now, we still may want to always backtick in case someone has some evil project/dataset/table name. But we could also wait to see it. Not hard to add. WDYT?

Reply:
On further digging, looks like that was an older requirement that they've removed - ignore.